streams/src/main/java/org/apache/kafka/streams/state/internals/StreamThreadStateStoreProvider.java

```diff
@@ -119,7 +121,7 @@ private TaskId createKeyTaskId(final String storeName, final Integer partition)
                 return new TaskId(topicGroup.getKey(), partition);
             }
         }
-        throw new InvalidStateStoreException("Cannot get state store " + storeName + " because the requested partition " +
+        throw new InvalidStateStorePartitionException("Cannot get state store " + storeName + " because the requested partition " +
```

Comment:
I guess here is not a exception about invalid partition and not sure which type of InvalidStateStoreException we should use. Using InvalidStateStorePartitionException here maybe not a correct choice. 
Any comment?